src/scripting/lua_game.cc

```diff
@@ -902,6 +903,68 @@ int LuaPlayer::get_buildings(lua_State* L) {
 	return 1;
 }
 
+/* RST
+   .. method:: get_constructionsites(which)
+
+      which can be either a single name or an array of names. In the first
+      case, the method returns an array of all Constructionssites that the player has
+      of this kind. If which is an array, the function returns a table of
+      (name,array of buildings) pairs.
+
+      :type which: name of constructionsites building or array of building names
+      :rtype which: :class:`string` or :class:`array`
+      :returns: information about the players constructionsites
+      :rtype: :class:`array` or :class:`table`
+*/
+int LuaPlayer::get_constructionsites(lua_State* L) {
```

Comment:
This function is 99% identical to `LuaPlayer::get_buildings`. Please pull the common code out into a new helper function `LuaPlayer::do_get_buildings(lua_State* L, const bool csites)` and change the two Lua hooks into one-liners that invoke the helper.

Reply:
done